### PR TITLE
MinIO: Extend test coverage #4707

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4707

Last item was not covered fully (`Cover some potentially risky options (risky in the native mode) from CamelMinioContentMD5 , CamelMinioETag, CamelMinioStorageClass, CamelMinioVersionId, CamelMinioCannedAcl`),
because the code responsible for this feature in Camel component is just very simple setter for MinIO.

### CODE REVIEW

File: integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java [hunk 1]
```diff
@@ -30,76 +34,169 @@
...
+    @Inject
+    CamelContext camelContext;
+
+    @Path("/consumerWithClientCreation/{endpoint}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response consumerWithClientCreation(@PathParam("endpoint") String endpoint) {
+        MinioComponent minioComponent = camelContext.getComponent("minio", MinioComponent.class);
+        MinioClient client = minioComponent.getConfiguration().getMinioClient();
+        minioComponent.getConfiguration().setMinioClient(null);
```
Comment (jamesnetherton) [inline integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java:74]: May be better to have a dedicated component instance for this, then you don't need to mess with the original `minio` instance.

E.g

```
@Produces
@Named("minioCamel")
MinioComponent camelMinioComponent() {
    // Code to set null client, autowired disabled etc
}
```

File: integration-tests/minio/src/test/java/org/apache/camel/quarkus/component/minio/it/MinioTestResource.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ public class MinioTestResource implements QuarkusTestResourceLifecycleManager {
     public static final String CONTAINER_SECRET_KEY = "MINIO_SECRET_KEY";
     private final String CONTAINER_IMAGE = "minio/minio:RELEASE.2020-12-03T05-49-24Z";
```
Comment (jamesnetherton) [inline integration-tests/minio/src/test/java/org/apache/camel/quarkus/component/minio/it/MinioTestResource.java:31]: Might be good to update the container image. Maybe to this:

https://github.com/quarkiverse/quarkus-minio/blob/main/minio-client/deployment/src/main/java/io/quarkiverse/minio/client/deployment/devservices/MinioDevServicesBuildTimeConfig.java#L29

**Comment by zbendhiba** [comment]
@JiriOndrusek Since you are working on Minio, maybe you could take a look at test failing while upgrading the dependency. cf https://github.com/apache/camel-quarkus/pull/4831

File: integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java (hunk 1 above)
Comment (JiriOndrusek) [reply integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java:74]: you are right, that would be "cleaner" solution, thanks for the hint, I'll change that

Comment (JiriOndrusek) [reply integration-tests/minio/src/main/java/org/apache/camel/quarkus/component/minio/it/MinioResource.java]: done

File: integration-tests/minio/src/test/java/org/apache/camel/quarkus/component/minio/it/MinioTestResource.java (hunk 2 above)
Comment (JiriOndrusek) [reply integration-tests/minio/src/test/java/org/apache/camel/quarkus/component/minio/it/MinioTestResource.java]: done

**Comment by jamesnetherton** [comment]
Is this good to merge? I guess we have to delay the Minio upgrade until the okhttp issue is fixed.

**Comment by JiriOndrusek** [comment]
It is good to merge. This PR extends coverage on current version.

Upgrade of minio might not happen because of https://github.com/quarkiverse/quarkus-minio/issues/220

**Pull request merged**